context/src/main/java/io/grpc/Context.java

```diff
@@ -719,8 +719,9 @@ public void detach(Context toAttach) {
       uncancellableSurrogate.detach(toAttach);
     }
 
+    // Visible for testing
     @Override
-    public boolean isCurrent() {
+    boolean isCurrent() {
```

Comment:
Agreed that this may cause pain for users, will deprecate this instead and add a javadoc about what's happening here.